src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -214,31 +397,368 @@ public void log(Logger logger) {
             logger.info("Bulk requests resolution is {} during request auditing.", resolveBulkRequests ? "enabled" : "disabled");
             logger.info("Index resolution is {} during request auditing.", resolveIndices ? "enabled" : "disabled");
             logger.info("Sensitive headers auditing is {}.", excludeSensitiveHeaders ? "enabled" : "disabled");
-            logger.info("Auditing requests from {} users is disabled.", ignoredAuditUsersMatcher);
-        }
-
-        @Override
-        public String toString() {
-            return "Filter{" +
-                    "isRestApiAuditEnabled=" + isRestApiAuditEnabled +
-                    ", disabledRestCategories=" + disabledRestCategories +
-                    ", isTransportApiAuditEnabled=" + isTransportApiAuditEnabled +
-                    ", disabledTransportCategories=" + disabledTransportCategories +
-                    ", resolveBulkRequests=" + resolveBulkRequests +
-                    ", logRequestBody=" + logRequestBody +
-                    ", resolveIndices=" + resolveIndices +
-                    ", excludeSensitiveHeaders=" + excludeSensitiveHeaders +
-                    ", ignoredAuditUsers=" + ignoredAuditUsersMatcher +
-                    ", ignoreAuditRequests=" + ignoredAuditRequestsMatcher +
-                    '}';
+            logger.info("Auditing requests from {} users is disabled.", ignoredAuditUsers);
+        }
+    }
+
+    public static class Compliance {
+        @JsonProperty(value = Key.ENABLED)
+        private boolean complianceEnabled = true;
+        @JsonProperty(value = Key.INTERNAL_CONFIG_ENABLED)
+        private boolean internalConfigEnabled = true;
```

Comment:
Yes internal_config_enalbed logs changes to security index. Since audit logging is also stored in the index, changes will be logged. However if disabled then config changes are not recorded. Will whitelist to record audit config changes always.

Reply:
Will whitelist to record audit config changes always. --> why do we need to do whitelist? We should not treat auditlogging config specially. It should be same as other user/action/roles/role-mapping, cert-Dn etc.. 

It should be enabled (Default: True) and customer should be able to disable if they want.